index.php

```diff
@@ -744,229 +472,811 @@ function addSearchResults(xmlHttp) {
+                        }
+                    }),
+                    properties: {
+                        name: "seamarks",
+                        layerId: 3,
+                        cookieKey: "SeamarkLayerVisible",
+                        checkboxId: "checkLayerSeamark",
+                    }
+                });
+
+                layer_seamark.on("change:visible", (evt) => {
+                    updateCheckboxAndCookie(evt.target);
+                });
+
                 // Sport
-                layer_sport = new OpenLayers.Layer.TMS("Sport", "https://tiles.openseamap.org/sport/",
-                    { layerId: 4, numZoomLevels: 19, type: 'png', getURL:getTileURL, isBaseLayer:false, visibility: false, displayOutsideMaxExtent:true});
+                // old definition
+                // layer_sport = new OpenLayers.Layer.TMS("Sport", "https://tiles.openseamap.org/sport/",
+                //     { layerId: 4, numZoomLevels: 19, type: 'png', getURL:getTileURL, isBaseLayer:false, visibility: false, displayOutsideMaxExtent:true});
+                layer_sport = new ol.layer.Tile({
+                    visible: false,
+                    maxZom: 19,
+                    properties: {
+                        name: 'Sport',
+                        layerId: 4,
+                        checkboxId: "checkLayerSport",
+                        cookieKey: "SportLayerVisible",
+                    },
+                    source: new ol.source.XYZ({
+                        url: "https://tiles.openseamap.org/sport/{z}/{x}/{y}.png",
+                        tileUrlFunction: function(coordinate) {
+                            return getTileUrlFunction("https://tiles.openseamap.org/sport/", 'png', coordinate);
+                            // return "https://tiles.openseamap.org/seamark/" + coordinate[0] + '/' +
+                            //     coordinate[1] + '/' + (-coordinate[2] - 1) + '.png';
+                        }
+                        
+                    }),
+                });
+                layer_sport.on("change:visible", (evt) => {
+                    updateCheckboxAndCookie(evt.target);
+                });
+
                 //GebcoDepth
-                layer_gebco_deeps_gwc = new OpenLayers.Layer.WMS("gebco_2021", "https://depth.openseamap.org/geoserver/gwc/service/wms",
-                    {layers: "gebco2021:gebco_2021", format:"image/png"},
-                    { layerId: 6, isBaseLayer: false, visibility: false, opacity: 0.8});
+                // old definition
+                // layer_gebco_deeps_gwc = new OpenLayers.Layer.WMS("gebco_2021", "https://depth.openseamap.org/geoserver/gwc/service/wms",
+                //     {layers: "gebco2021:gebco_2021", format:"image/png"},
+                //     { layerId: 6, isBaseLayer: false, visibility: false, opacity: 0.8});
+                layer_gebco_deeps_gwc = new ol.layer.Tile({
```

Comment:
The Gebco layer seems broken. Best I can determine, the new code leads to parameters WIDTH and HEIGTH set to values that are not 256 (the map server seems to only support 256). On my machine, WIDHT=HEIGHT=307.

Reply:
Works for me . It seems it's broken only when the device has pxelRatio > 1.

Reply:
Thanks for pointing this out , I have deactivated the hidpi functionnaility for all TileWMS source.